src/reducer.js

```diff
@@ -93,6 +95,35 @@ const initialState = {
   tempPin: null,
   isShowingPinScreen: false,
   walletStartState: WALLET_STATUS.LOADING,
+  pushNotification: {
+    /**
+     * deviceId {string} device id for push notification
+     */
+    deviceId: '',
+    /**
+     * apiStatus {string} status of the push notification api request
+     */
+    apiStatus: PUSH_API_STATUS.READY,
+    /**
+     * enabled {boolean} if user has enabled push notification
+     */
+    enabled: false,
+    /**
+     * showAmountEnabled {boolean} if user has enabled the option to show amount
+     * of token balance on the push notification
+     */
+    showAmountEnabled: false,
+    /**
+     * hasBeenEnabled {boolean} if user has enabled push notification before
+     * this is used to show the terms and conditions modal only the first time
+     * user enables push notification
+     */
+    hasBeenEnabled: false,
```

Comment:
If we decide to remove the modal, then we can remove this state here

Reply:
Actually, this is used to trigger the registration when the user enables the push notification for the first time. When the user enables it for the second time if any, the update function is triggered.